### PR TITLE
Add smithy-templates file

### PR DESCRIPTION
**Pull request by hpmellema**
### Description of changes
Adds a `smithy-templates.json` file to the repository so examples can be copied using the `smithy init` CLI tool.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-templates.json [hunk 1]
```diff
@@ -0,0 +1,125 @@
+{
+  "name": "smithy-gradle-plugin",
+  "templates": {
+    "include-in-sourceset": {
```
Comment (mtdowling) [inline smithy-templates.json:4]: Should this be plural?
```suggestion
    "includes-in-sourceset": {
```

File: smithy-templates.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
+{
```
Comment (mtdowling) [inline smithy-templates.json:1]: Should each description say "Uses Gradle to"? Or is it apparent from the repo name?

File: smithy-templates.json [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      "path": "examples/jar-plugin/source-projection",
+      "include": [
+        "gradle/",
+        "gradlew",
+        "gradlew.bat",
+        ".gitignore"
+      ]
+    }
+  }
+}
```
Comment (mtdowling) [inline smithy-templates.json:125]: nit: missing newline

File: smithy-templates.json (hunk 2 above)
Comment (hpmellema) [reply smithy-templates.json:1]: I think this is apparent for the repo name.

**Review by srchase — APPROVED** [review]

**Pull request merged**